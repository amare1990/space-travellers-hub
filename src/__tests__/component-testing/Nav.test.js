import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavBar from '../../pages/Nav';
import store from '../../redux/configureStore';

test('Testing if NavBar component renders correctly', () => {
  const navBarSnapshot = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(navBarSnapshot).toMatchSnapshot();
});
