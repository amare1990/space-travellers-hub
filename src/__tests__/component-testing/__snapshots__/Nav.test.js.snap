// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if NavBar component renders correctly 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="logo-and-title"
    >
      <img
        alt="Logo"
        className="logo"
        src="planet.png"
      />
      <h1
        className="title"
      >
        Space Traveler\`s Hub
      </h1>
    </div>
    <ul
      className="navbar"
    >
      <li
        className="link-item"
      >
        <a
          aria-current="page"
          className="item active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="link-item"
      >
        <a
          className="item"
          href="/Missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="link-item"
      >
        <a
          className="item"
          href="/Profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </header>,
  <div
    className="horizontal-bar"
  />,
]
`;
